src/main/java/io/anserini/index/IndexReaderUtils.java

```diff
@@ -791,4 +672,77 @@ public static String convertLuceneDocidToDocid(IndexReader reader, int docid) {
     }
     return indexStats;
   }
+
+  /**
+   * Returns {@code FieldInfo} for indexed fields.
+   *
+   * @param reader index reader
+   * @return map from name of field to its {@code FieldInfo}
+   */
+  public static Map<String, FieldInfo> getFieldInfo(IndexReader reader) {
+    Map<String, FieldInfo> fields = new HashMap<>();
+
+    FieldInfos fieldInfos = FieldInfos.getMergedFieldInfos(reader);
+    for (FieldInfo fi : fieldInfos) {
+      fields.put(fi.name, fi);
+    }
+
+    return fields;
+  }
+
+  /**
+   * Returns string summary of {@code FieldInfo} for indexed fields.
+   *
+   * @param reader index reader
+   * @return map from name of field to its {@code FieldInfo} string summary
+   */
+  public static Map<String, String>  getFieldInfoDescription(IndexReader reader) {
+    Map<String, String> description = new HashMap<>();
+
+    FieldInfos fieldInfos = FieldInfos.getMergedFieldInfos(reader);
+    for (FieldInfo fi : fieldInfos) {
+      description.put(fi.name, "(" + "indexOption: " + fi.getIndexOptions() + ", hasVectors: " + fi.hasVectors() + ")");
+    }
+
+    return description;
+  }
+
+  // This is needed by src/main/python/run_regression.py
+
+  public static final class Args {
+    @Option(name = "-index", metaVar = "[Path]", required = true, usage = "index path")
+    String index;
+
+    @Option(name = "-stats", usage = "print index statistics")
+    boolean stats;
+  }
+
+  public static void main(String[] argv) throws Exception{
+    Args args = new Args();
+    CmdLineParser parser = new CmdLineParser(args, ParserProperties.defaults().withUsageWidth(90));
+    try {
+      parser.parseArgument(argv);
+    } catch (CmdLineException e) {
+      System.err.println(e.getMessage());
+      parser.printUsage(System.err);
+      return;
+    }
+
+    Directory directory = FSDirectory.open(new File(args.index).toPath());
+    IndexReader reader = DirectoryReader.open(directory);
+
+    if (args.stats) {
+      Terms terms = MultiTerms.getTerms(reader, IndexArgs.CONTENTS);
+
+      System.out.println("Index statistics");
+      System.out.println("----------------");
+      System.out.println("documents:             " + reader.numDocs());
+      System.out.println("documents (non-empty): " + reader.getDocCount(IndexArgs.CONTENTS));
+      System.out.println("unique terms:          " + terms.size());
+      System.out.println("total terms:           " + reader.getSumTotalTermFreq(IndexArgs.CONTENTS));
+    }
```

Comment:
Why not obtain stats from `getIndexStats` and print those? Reduces duplicate code

Reply:
oh, good call! 👍